src/libraries/Common/src/System/Net/WebSockets/WebSocketValidate.cs

```diff
@@ -116,22 +138,20 @@ internal static void ValidateCloseStatus(WebSocketCloseStatus closeStatus, strin
             }
         }
 
-        internal static void ValidateArraySegment(ArraySegment<byte> arraySegment, string parameterName)
+        internal static void ValidateArraySegment(ArraySegment<byte> arraySegment, [CallerArgumentExpression(nameof(arraySegment))] string? parameterName = null)
         {
             Debug.Assert(!string.IsNullOrEmpty(parameterName), "'parameterName' MUST NOT be NULL or string.Empty");
 
-            if (arraySegment.Array == null)
-            {
-                throw new ArgumentNullException(parameterName + "." + nameof(arraySegment.Array));
-            }
-            if (arraySegment.Offset < 0 || arraySegment.Offset > arraySegment.Array.Length)
-            {
-                throw new ArgumentOutOfRangeException(parameterName + "." + nameof(arraySegment.Offset));
-            }
-            if (arraySegment.Count < 0 || arraySegment.Count > (arraySegment.Array.Length - arraySegment.Offset))
-            {
-                throw new ArgumentOutOfRangeException(parameterName + "." + nameof(arraySegment.Count));
-            }
+            string arrayParameterName = parameterName + "." + nameof(arraySegment.Array);
```

Comment:
This is not an acceptable change. It is now allocating a string regardless of whether the argument is nulll.

Reply:
Ah right! Thanks for spotting. Will revert this.

Reply:
Custom interpolated string handler short-circuit would be nice in the new throw helpers for this 😅

Reply:
> Not worth it probably.

No. There would still be additional checks performed. In the production path where the argument should _never_ be null, we want the overhead to be nothing more than a single comparison/branch.